src/snowflake/snowpark/mock/_functions.py

```diff
@@ -390,6 +390,15 @@ def mock_covar_pop(column1: ColumnEmulator, column2: ColumnEmulator) -> ColumnEm
     )
 
 
+@patch("array_agg")
+def mock_array_agg(column: ColumnEmulator, is_distinct: bool) -> ColumnEmulator:
+    columns_data = ColumnEmulator(column.unique()) if is_distinct else column
+    return ColumnEmulator(
+        data=[list(columns_data.dropna())],
+        sf_type=ColumnType(ArrayType(StringType()), False),
```

Comment:
dumb question, why it's StringType not the data type of the input column?

Reply:
I can actually probably remove that. The old default was for it to be string type, but I think with some of my more recent changes it should be able to adapt without specifying. That way when structured type support is done rolling out it won't need to be updated her.